File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        left = [1] * len(nums)
        right = [1] * len(nums)
        output = [0] * len(nums)
        l = 1
        r = 1

        for i in range(len(nums)):
            j = len(nums) - i - 1
            left[i] *= l
            l *= nums[i]
            right[j] *= r
            r *= nums[j]

        for i in range(len(left)):
            output[i] = left[i] * right[i]
        
        return output
        

File: Data Structures & Algorithms/lowest-common-ancestor-in-binary-search-tree/submission-1.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public TreeNode lowestCommonAncestor(TreeNode root, TreeNode p, TreeNode q) {
        TreeNode result;
        if (root.val >= q.val && root.val <= p.val || root.val <= q.val && root.val >= p.val) {
                result = root;
            }
        else {
            if (p.val > root.val) {
                result = lowestCommonAncestor(root.right, p, q);
            } else {
                result = lowestCommonAncestor(root.left, p, q);
            }
        }
        
            
        return result;
    }
}